extensions/amp-iframe/1.0/component.js

```diff
@@ -28,6 +34,54 @@ export function Iframe({
   const dataRef = useRef(null);
   const isIntersectingRef = useRef(null);
   const containerRef = useRef(null);
+  const observerRef = useRef(null);
+
+  const viewabilityCb = (entries) => {
+    const iframe = iframeRef.current;
+    if (!iframe) {
+      return;
+    }
+    postMessage(
+      iframe,
+      MessageType.INTERSECTION,
+      dict({'changes': entries.map(cloneEntryForCrossOrigin)}),
+      '*'
```

Comment:
@jridgewell any ideas how to include the origin here? One way I can think of is:
```
const {origin} = new URL(src, win.location.origin) // win.location.origin to accommodate relative URLs
```

Reply:
Because of redirect, you can never know the real origin at this point. You could keep the origins of the windows that send you the `SEND_INTERSECTIONS` event (`ev.origin`), then send to those origins.

Reply:
Done. Thanks!